applications/tari_merge_mining_proxy/src/proxy.rs

```diff
@@ -605,20 +642,41 @@ impl InnerService {
             }
         }
 
-        for monerod_url in self.config.monerod_url.iter() {
-            let uri = format!("{}{}", monerod_url, uri.path()).parse::<Url>()?;
-            match reqwest::get(uri.clone()).await {
-                Ok(_) => {
-                    let mut lock = self.last_available_server.write().expect("Write lock should not fail");
-                    *lock = Some(monerod_url.to_string());
-                    info!(target: LOG_TARGET, "Monerod server available: {:?}", uri.clone());
-                    return Ok(uri);
-                },
-                Err(_) => {
-                    warn!(target: LOG_TARGET, "Monerod server unavailable: {:?}", uri);
-                    continue;
-                },
+        let current_url = {
+            let lock = self
+                .last_available_server
+                .read()
+                .expect("Read lock should not fail")
+                .clone();
+            match lock {
+                Some(url) => url,
+                None => "".to_string(),
             }
+        };
+        let mut iter = self.config.monerod_url.iter();
+        let _ = iter.find(|&x| x == &current_url);
+        let mut count = 0;
+        loop {
+            count += 1;
+            if count > self.config.monerod_url.len() * 2 {
```

Comment:
times 2? Is the intention to go through this list twice?

Reply:
See previous comment - removed the loop